.sun-ci.yml

```
@@ -0,0 +1,90 @@
+default:
+  workspace: shared
+
+stages:
+  - build
+  - check
+
+jobs:
+  - name: Build Laravel
+    stage: build
+    image: ghcr.io/avengerscodelovers/php:8.2-ci
+    services:
+      - image: mysql:8.0
+        name: mysql
+        environment:
+          MYSQL_DATABASE: app_db
+          MYSQL_USER: app_user
+          MYSQL_PASSWORD: secret
+          MYSQL_ROOT_PASSWORD: root
+    environment:
+      APP_ENV: testing
+      COMPOSER_MEMORY_LIMIT: 2G
+      COMPOSER_ALLOW_SUPERUSER: "1"
+    cache:
+      - key: composer_vendor_$CI_BRANCH
+        paths:
+          - vendor
+    script:
+      - cp .env.testing.example .env
+      - composer install
+      - php artisan key:generate
+      - php artisan migrate
+      - php artisan route:cache
+
+  # - name: Check GKC Hashcode
+  #   stage: check
+  #   image: bash:latest
+  #   workspace: false
+  #   before_script:
+  #     # For command tput, used by gkc_util_hashcode_script.sh
+  #     - apk add ncurses
+  #   environment:
+  #     TERM: xterm
+  #   script:
+  #     - bash ./gkc_util_hashcode_script.sh
+  #   only:
+  #     events:
+  #       - pull_request
+
+  - name: Static Code Analysis
+    stage: check
+    image: ghcr.io/avengerscodelovers/php:8.2-ci
+    environment:
+      COMPOSER_MEMORY_LIMIT: 2G
+      COMPOSER_ALLOW_SUPERUSER: "1"
+    script:
+      - COMPOSER_PROCESS_TIMEOUT=600 composer phpstan
+
+  - name: PHP Coding Convention
+    stage: check
+    image: ghcr.io/avengerscodelovers/php:8.2-ci
+    environment:
+      COMPOSER_MEMORY_LIMIT: 2G
+      COMPOSER_ALLOW_SUPERUSER: "1"
+    script:
+      - composer cs
+
+  - name: PHP Test Coverage and Codecov
+    stage: check
+    image: ghcr.io/avengerscodelovers/php:8.2-ci
+    services:
+      - image: mysql:8.0
+        name: mysql
+        environment:
+          MYSQL_DATABASE: app_db
+          MYSQL_USER: app_user
+          MYSQL_PASSWORD: secret
+          MYSQL_ROOT_PASSWORD: root
```

```suggestion
      <<: *commonServices
```
Thử xem có tận dụng lại được block giống nhau bên trong file yml xem seo
https://gist.github.com/vitaLee/7803420

---

Cái này hay đấy https://joshdevlin.com/blog/yaml-repeating-sections/